### DISCUSSION TITLE
How to add support for new custom 4x6 keyboard?

### DISCUSSION THREAD

**Discussion by wespiard** [manna-harbour/miryoku#102] [Q&A] [answered]
I just finished my first custom keyboard that is 4x6+3. I flashed it with a single Colemak layer just to make sure all of the keys worked. Now I'd like to try to get Miryoku working for the base 3x5+3 keys, and then add numbers to the top row and some misc. keys to the outer column just for gaming purposes.

If there is already any documentation on how to do this somewhere, that would be great too. I just couldn't find it very quickly.

Thanks!

**Comment by manna-harbour** [comment, marked as answer]
Different physical layouts are handled by mapping macros. There's not much documentation yet, but there are plenty of examples. 

For Miryoku QMK see https://github.com/manna-harbour/miryoku_qmk/tree/miryoku/users/manna-harbour_miryoku#keyboards. For 4x6+3 you can use https://github.com/manna-harbour/miryoku_qmk/blob/miryoku/keyboards/bastardkb/scylla/keymaps/manna-harbour_miryoku/config.h and delete https://github.com/manna-harbour/miryoku_qmk/blob/e9dfa7d007e9655ebe6ef7c22bc6112eb228782a/keyboards/bastardkb/scylla/keymaps/manna-harbour_miryoku/config.h#L22 and change https://github.com/manna-harbour/miryoku_qmk/blob/e9dfa7d007e9655ebe6ef7c22bc6112eb228782a/keyboards/bastardkb/scylla/keymaps/manna-harbour_miryoku/config.h#L16.

For Miryoku ZMK see  https://github.com/manna-harbour/miryoku_zmk#mapping-macros. 

To map the unused keys, you can do it directly in the mapping macro instead of the `XXX`s, or to do it at the keymap level see https://github.com/manna-harbour/miryoku/discussions/85.

**Reply by wespiard** [reply]
Perfect, I'll give it a shot. Looks a lot easier than I was imagining.

**Reply by wespiard** [reply]
I think I got it working. Took me a while to realize I needed to add a folder to my keymaps directory. I just assumed that there was some more magic with the userspace than there actually was...
